### ISSUE TITLE
Groupby_dynamic without Holes

### ISSUE THREAD

**Issue by MariusMerkleQC** [pola-rs/polars#8831]
### Problem description

If I use the `groupby_dynamic` to count the number of entries for every hour, only non-zero entries are counted. For example, if I have entries at "2023-05-13 06:05:00", "2023-05-13 07:05:00", and "2023-05-13 09:05:00", then the result will have a `1` for "2023-05-13 06:00:00", "2023-05-13 07:00:00", and "2023-05-13 09:00:00". I'd like to also have an entry with `0` for "2023-05-13 08:00:00".

In other words, is it possible to add an optional argument `include_all` to the `groupby_dynamic` function that includes a count for all values in between the first and last time?

**Comment by cmdlineluser** [comment]
It sounds like you want to [`.upsample()`](https://pola-rs.github.io/polars/py-polars/html/reference/dataframe/api/polars.DataFrame.upsample.html) before grouping?

```
df = pl.from_repr("""
┌─────────────────────┬─────┐
│ date                ┆ id  │
│ ---                 ┆ --- │
│ datetime[μs]        ┆ i64 │
╞═════════════════════╪═════╡
│ 2023-05-13 06:05:00 ┆ 1   │
│ 2023-05-13 07:05:00 ┆ 2   │
│ 2023-05-13 09:05:00 ┆ 3   │
└─────────────────────┴─────┘
""")

df.upsample(time_column="date", every="1h")

# shape: (4, 2)
# ┌─────────────────────┬──────┐
# │ date                ┆ id   │
# │ ---                 ┆ ---  │
# │ datetime[μs]        ┆ i64  │
# ╞═════════════════════╪══════╡
# │ 2023-05-13 06:05:00 ┆ 1    │
# │ 2023-05-13 07:05:00 ┆ 2    │
# │ 2023-05-13 08:05:00 ┆ null │
# │ 2023-05-13 09:05:00 ┆ 3    │
# └─────────────────────┴──────┘

```

**Comment by ritchie46** [comment]
Exactly what @cmdlineluser says. A groupby should not invent groups because that fits a nice regular pattern. If you want a regular pattern, you must first create that.

**Comment by MariusMerkleQC** [comment]
Great, that works, thank you!
